.gitlab/llnl-ci-conda.yml

```diff
@@ -0,0 +1,126 @@
+variables:
+  WORK_DIR: ${SDK_WORK_DIR}/conda
+  REQUIREMENTS: .gitlab/conda-requirements.yml
+  RADICAL_PILOT_DBURL: ${MONGODB_CONNECTION_STRING}?tlsAllowInvalidCertificates=true
+  TEST_TYPE: "conda"
+
+stages:
+  - setup
+  - build
+  - test
+  - cleanup
+
+.final_steps:
+  script: &finalize
+    - chgrp -fR exaworks $WORK_DIR || true
+    - chmod -fR 02770 $WORK_DIR    || true
+
+conda-env-setup:
+  stage: setup
+  parallel:
+    matrix:
+      - HOST : [ "quartz" ]
+        CONDA_ENV_NAME: [ "quartz-env" ]
+        RUNNER_TYPE: "shell"
+      - HOST : [ "ruby" ]
+        CONDA_ENV_NAME: [ "ruby-env" ]
+        RUNNER_TYPE: "shell"
+      # - HOST : [ "lassen" ]
+      #   ENVIRONMENT: [ "BATCH" ]
+      #   CONDA_ENV_NAME: [ "lassen-env" ]
+      #   RUNNER_TYPE: "batch"
+  tags:
+    - $HOST
+    - $RUNNER_TYPE
+  script:
+    - test -d ${WORK_DIR}/${CONDA_ENV_NAME} && exit 0
+    - mkdir -p ${WORK_DIR}
+    - eval "$(conda shell.posix hook)"
+    - conda create --prefix ${WORK_DIR}/${CONDA_ENV_NAME} python=3.7 -y
+    - conda activate ${WORK_DIR}/${CONDA_ENV_NAME}
+
+conda-build:
+  stage: build
+  parallel:
+    matrix:
+      - HOST : [ "quartz" ]
+        CONDA_ENV_NAME: [ "quartz-env" ]
+        RUNNER_TYPE: "shell"
+      - HOST : [ "ruby" ]
+        CONDA_ENV_NAME: [ "ruby-env" ]
+        RUNNER_TYPE: "shell"
+      # - HOST : [ "lassen" ]
+      #   ENVIRONMENT: [ "BATCH" ]
+      #   CONDA_ENV_NAME: [ "lassen-env" ]
+      #   RUNNER_TYPE: "batch"
+  tags:
+    - $HOST
+    - $RUNNER_TYPE
+  script:
+    - eval "$(conda shell.posix hook)"
+    - conda activate ${WORK_DIR}/${CONDA_ENV_NAME}
+    - conda env update --file ${REQUIREMENTS}
+    - *finalize
+
+conda-tests:
+  stage: test
+  needs:
+    - job: conda-env-setup
+  parallel:
+    matrix:
+      - HOST : [ "quartz" ]
+        CONDA_ENV_NAME: [ "quartz-env" ]
+        RUNNER_TYPE: "batch"
+      - HOST : [ "ruby" ]
+        CONDA_ENV_NAME: [ "ruby-env" ]
+        RUNNER_TYPE: "batch"
+      # - HOST : [ "lassen" ]
+      #   ENVIRONMENT: [ "BATCH" ]
+      #   CONDA_ENV_NAME: [ "lassen-env" ]
+      #   RUNNER_TYPE: "batch"
+  tags:
+    - $HOST
+    - $RUNNER_TYPE
+  script:
+    - eval "$(conda shell.posix hook)"
+    - conda activate ${WORK_DIR}/${CONDA_ENV_NAME}
+    - export run_id=${CI_PIPELINE_ID}
+    - export branch=${CI_COMMIT_BRANCH}
+    - export url=${TESTING_HOST}
+    - export test=${TEST_TYPE}
+    - export location="llnl-$HOST"
+    - export contact="arambula2@llnl.gov"
+    - export imnumber="LLNL-MI-834241"
+    - python3 ci/tests/test.py -s
+    - for TEST in flux parsl rp swift-t parsl-flux rp-flux; do
+        python3 ci/tests/test.py -n $TEST -c "bash ci/tests/${TEST}/test.sh";
+      done
+    - python3 ci/tests/test.py -e
+
+.conda-env-cleanup: &conda-env-cleanup
+  stage: cleanup
+  tags:
+    - quartz
+    - shell
+  script:
+    - cd ${SDK_WORK_DIR}
+    - rm -rf conda
```

Comment:
these lines are not needed
(we later, with other PR, will add reporting to Dashboard if build stage is failed)

Reply:
done